src/document-context/measure-custom-elements.ts

```diff
@@ -167,61 +174,42 @@ interface Console {
     const originalDefine = window.customElements.define;
     const boundDefine: typeof originalDefine =
         originalDefine.bind(window.customElements);
-    let elementCounter = 0;
 
+    const counterMap = new Map<string, number>();
+    window['HTMLElement'] = class extends HTMLElement {
+      constructor() {
+        super();
+        const tagName = this.tagName.toLowerCase();
+        let counter = counterMap.get(tagName) || 0;
+        this[idSymbol] = counter;
+        counterMap.set(tagName, counter + 1);
+      }
+    };
     function wrappedDefineElement(
-        tagName: string,
-        constructor: CustomElementV1Constructor): CustomElementV1Constructor {
-      let wrappedConstructor: CustomElementV1Constructor =
-          class extends constructor {
-        constructor(...args: any[]) {
-          const counter = elementCounter++;
-          const startMark = `${prefix}start created ${tagName} ${counter}`;
-          const endMark = `${prefix}end created ${tagName} ${counter}`;
-          const measure = `${prefix}created ${tagName} ${counter}`;
-
-          window.performance.mark(startMark);
-          try {
-            super(...args);
-          } finally {
-            window.performance.mark(endMark);
-            window.performance.measure(measure, startMark, endMark);
-          }
-          this[idSymbol] = counter;
-        }
-
-        connectedCallback() {
-          const connectedCallback = super.connectedCallback || (() => null);
-          makeMeasurement(
-              'connected', tagName, this[idSymbol], connectedCallback);
-        }
-
-        disconnectedCallback() {
-          const disconnectedCallback =
-              super.disconnectedCallback || (() => null);
-          makeMeasurement(
-              'disconnected', tagName, this[idSymbol], disconnectedCallback);
-        }
-
-        attributeChangedCallback() {
-          const attributeChangedCallback =
-              super.attributeChangedCallback || (() => null);
-          makeMeasurement(
-              'attributeChanged', tagName, this[idSymbol],
-              attributeChangedCallback);
+        tagName: string, constructor: CustomElementV1Constructor): void {
+      let proto = constructor.prototype;
+      function wrapCustomElementCallback(
+          name: CallbackShorthand, fullName: Ce1CallbackName) {
+        const original = proto[fullName];
+        if (!original) {
+          return;
         }
+        proto[fullName] = function(this: any) {
+          const counter: number = this[idSymbol];
 
-        _propertySetter() {
-          const propertyChangedCallback =
-              super['_propertySetter'] || (() => null);
-          makeMeasurement(
-              'data', tagName, this[idSymbol], propertyChangedCallback);
-        }
+          return makeMeasurement(
+              name, tagName, counter, () => original.apply(this, arguments));
```

Comment:
Yeah, we deliberately simplify the different data binding APIs that we track to estimate databinding time down to a single `data` measurement. Partly because I'm not sure that we're not going to need to modify it to track other databinding functions.